portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -35,19 +40,38 @@
 public class DeleteDataServlet extends HttpServlet {
 
   private static final String QUOTE = "Quote";
+  private static final String USER_EMAIL = "userEmail";
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+    String userEmail = userService.getCurrentUser().getEmail(); 
+
+    DatastoreService datastore = getDatastoreServiceWithConsistency();
     Query query = new Query(QUOTE);
     PreparedQuery quotes = datastore.prepare(query);
 
+    // Only delete quotes that were submitted by the currently logged in user. 
     List<Key> keyList = new ArrayList<Key>();
     quotes.asIterable()
         .forEach((quoteEntity) -> {
-          keyList.add(quoteEntity.getKey());
+          if (quoteEntity.getProperty(USER_EMAIL).equals(userEmail)) {
+            keyList.add(quoteEntity.getKey());
+          } 
         });
 
     datastore.delete(keyList);
   }
+
+  /**
+   * Sets up strong consistency for datastore service. 
+   * This strong consistency ensures that freshness is more important than availability
+   * so that the most up-to-date data is returned and displayed on the page.
+   */ 
+  private DatastoreService getDatastoreServiceWithConsistency() { 
```

Comment:
`static` here as well. You could even extract this method out into a Util class or something, and have both your servlets use the same method.

Reply:
Moved to helper class Util.